app.js

```diff
@@ -224,11 +251,26 @@ class ScreenTester extends React.Component {
 			const backgroundElement = document.getElementById('background');
 			const pattern = prevState.patterns[prevIndex];
 			
-			backgroundElement.style.background = pattern;
+			// Clear any existing animation classes
+			backgroundElement.className = '';
+			
+			if (typeof pattern === 'string') {
+				// Regular pattern - apply as background
+				backgroundElement.style.background = pattern;
+			} else if (pattern.type === 'animation') {
+				// Animation pattern - apply class and background if specified
+				backgroundElement.className = pattern.class;
+				backgroundElement.style.background = pattern.background || '';
+			}
 			
 			if (document.fullscreenElement || document.webkitFullscreenElement || 
 				document.mozFullScreenElement || document.msFullscreenElement) {
-				document.body.style.background = pattern;
+				if (typeof pattern === 'string') {
+					document.body.style.background = pattern;
+				} else if (pattern.type === 'animation') {
+					document.body.className = (document.body.className.replace(/\b(bouncing-box|scrolling-horizontal|scrolling-vertical|scrolling-diagonal|flicker-slow|flicker-medium|flicker-fast)\b/g, '').trim() + ' ' + pattern.class).trim();
+					document.body.style.background = pattern.background || '';
+				}
```

Comment:
this few lines of code are repeated, please make it reusable to improve code maintainability.